python/lsst/meas/extensions/shapeHSM/_hsm_higher_moments.py

```diff
@@ -361,15 +366,25 @@ def __init__(self, config, name, schema, metadata, logName=None):
 
     def measure(self, record, exposure):
         # Docstring inherited.
-        M = np.zeros((2, 2))
+
+        # If the PSF model is not available, this plugin would fail for all entries.
+        if (psf := exposure.getPsf()) is None:
+            raise measBase.FatalAlgorithmError("No PSF attached to the exposure.")
```

Comment:
If the `HsmPsfMoments` algorithm failed because of lack of PSF, I'd like this plugin to also fail fatally instead of attempting every single source/object and failing individually. So I moved this up here.